core/src/main/java/org/aerogear/mobile/core/metrics/publisher/NetworkMetricsPublisher.java

```diff
@@ -53,23 +54,27 @@ public void publish(@NonNull final Metrics[] metrics, @Nullable final Callback c
 
             json.put("data", data);
 
-            httpRequest.post(url, json.toString().getBytes());
+            httpRequest.post(url, json.toString().getBytes())
+                            .respondWith(new Responder<HttpResponse>() {
+                                @Override
+                                public void onResult(HttpResponse value) {
+                                    if (callback != null) {
```

Comment:
Same comment as before: I think it wouldn't be a bad idea to initialise the callback to a NOOP callback, so that no nullcheck will be necessary. WDYT?

Reply:
Callbacks as a pattern are next on my refactoring after we merge this PR.  So yes, but not in this PR :)